### PR TITLE
Emails Revamp

### PR DESCRIPTION
**Pull request by mihaic195**
What happened?
- [x] Rewrote all of the email templates 
- [x] Refactored `EventManager.ts`, `calendarClient.ts`, `videoClient.ts` and others.
- [x] Fixes email location issues (hopefully).

What still needs to be done but not deal-breaker:
- [ ] DRY up the email templates
- [ ] Move all email images on disk instead of remote (remaining images from footer and/or Cal.com logo).
- [ ] Some type errors on `lib/integrations` (outside the scope of this PR).

Testing needed on:
- [x] Zoom
- [x] Daily
- [x] Google Calendar
- [x] Address/Phone number
- [ ] Office 365
- [ ] Stripe
- [ ] Apple Calendar
- [ ] CalDav

### CODE REVIEW

File: lib/CalEventParser.ts [hunk 1]
```diff
@@ -1,151 +1,14 @@
...
-   */
-  public asRichEventPlain(): CalendarEvent {
-    const eventCopy: CalendarEvent = { ...this.calEvent };
-    eventCopy.description = this.getRichDescription();
-    eventCopy.location = this.getLocation();
-    return eventCopy;
-  }
-}
+export const getCancelLink = (calEvent: CalendarEvent) => {
+  return process.env.BASE_URL + "/cancel/" + getUid(calEvent);
```
Comment (alishaz-polymath) [inline lib/CalEventParser.ts:13]: Can we import and use `BASE_URL` from `lib/config/constants.ts` here as well?

Comment (zomars) [reply lib/CalEventParser.ts:13]: Yes @alishaz-polymath I was just testing if it works on vercel deployments and it seems like it is.

File: lib/calendarClient.ts [hunk 2]
```diff
@@ -683,27 +176,23 @@ const createEvent = async (
...
 ): Promise<EventResult> => {
-  const parser: CalEventParser = new CalEventParser(calEvent);
-  const uid = parser.getUid();
-  const richEvent: CalendarEvent = parser.asRichEventPlain();
-
+  const uid = getUid(calEvent);
   let success = true;
 
-  const updatedResult =
+  const updationResult =
```
Comment (zomars) [inline lib/calendarClient.ts:184]: Is this typo intentional? @mihaic195

File: pages/api/cancel.ts [hunk 3]
```diff
@@ -189,7 +191,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 
   await Promise.all([apiDeletes, attendeeDeletes, bookingReferenceDeletes]);
 
-  //TODO Perhaps send emails to user and client to tell about the cancellation
+  await sendCancelledEmails(evt);
```
Comment (zomars) [inline pages/api/cancel.ts:194]: Nice!

File: lib/calendarClient.ts (hunk 2 above)
Comment (mihaic195) [reply lib/calendarClient.ts:184]: We can update that.

File: test/lib/emails/invitation.test.ts [hunk 4]
```diff
@@ -1,27 +0,0 @@
...
-    teamName: "Calendar Lovers",
-    token: "invite-token",
-  } as Invitation;
-  const result = html(invitation);
-  expect(result).toContain(
-    `<br />${t("user_invited_you", { user: invitation.from, teamName: invitation.teamName })}<br />`
-  );
-  expect(result).toContain("/auth/signup?token=invite-token&");
-  expect(result).toContain(`${t("request_another_invitation_email", { toEmail: invitation.toEmail })}`);
-});
```
Comment (zomars) [inline test/lib/emails/invitation.test.ts:27]: We should design better tests in a followup PR

**Review by zomars — APPROVED** [review]
Awesome work @mihaic195 We should merge this and keep an eye for any missed cases.

**Comment by KATT** [comment]
> Awesome work @mihaic195 We should merge this and keep an eye for any missed cases.

+1

It's a big improvement, no nitpicking needed, it's a step in the right direction. Too large too be able to reliably spot any bugs or issues, so testing is needed

**Review by emrysal — APPROVED** [review]
All looks good, nothing to add that is already said, got some flows tested involving Daily and Zoom but it's a big all-encompassing to test everything

**Review by alishaz-polymath — APPROVED** [review]

**Pull request merged**